pantheon/src/main/java/tech/pegasys/pantheon/cli/subcommands/blocks/BlocksSubCommand.java

```diff
@@ -135,4 +132,100 @@ public void run() {
       }
     }
   }
+
+  /**
+   * blocks dump sub-command
+   *
+   * <p>Dump a specific block from storage
+   */
+  @Command(
+      name = "dump",
+      description = "This command dump a specific block from storage",
+      mixinStandardHelpOptions = true)
+  static class DumpSubCommand implements Runnable {
+    @SuppressWarnings("unused")
+    @ParentCommand
+    private BlocksSubCommand parentCommand; // Picocli injects reference to parent command
+
+    @Option(
+        names = "--number",
+        required = true,
+        paramLabel = MANDATORY_LONG_FORMAT_HELP,
+        description = "Number of the block to dump",
+        arity = "1..1")
+    private final Long blockNumber = null;
+
+    @Option(
+        names = "--to",
+        paramLabel = MANDATORY_FILE_FORMAT_HELP,
+        description = "File to write the block instead of standard output",
+        arity = "1..1")
+    private File dumpExportFile = null;
+
+    @Override
+    public void run() {
+
+      LOG.info("Runs dump sub command with block number : {}", blockNumber);
+
+      checkCommand(parentCommand);
+      checkNotNull(blockNumber);
+
+      Optional<MetricsService> metricsService = initMetrics(parentCommand);
+
+      try {
+
+        final MutableBlockchain blockchain =
```

Comment:
This subcommand should support exporting more than one block at a time.  To that end we will need another CLI option, either `--end-block` or `--block-count` (I prefer `--end-block`) to tell the export where to start.

`--end-block` would be optional.  If not specified I lean towards single block export instead of all block export, it's friendlier to hard drives.

If we don't have the start block, we should throw an exception for sure.  But if we don't find a block on the way to end block we probably should also throw an exception and clarify the export is a fraction of what was requested.  But don't delete the file.


If we don't have start block we throw the exception about not being able to find the start block.

If we run out of blocks before we hit a specifi

Reply:
I just made the requested changes here https://github.com/PegaSysEng/pantheon/pull/1641
--start-block is inclusive and --end-block is exclusive